nrfx HAL: Align with nrfx version 2.0.0

anangl opened the pull request:
Needed in order to integrate nrfx 2.0.0 into Zephyr.
Tested in https://github.com/zephyrproject-rtos/zephyr/pull/20436 by using [this temporary patch](https://github.com/zephyrproject-rtos/zephyr/pull/20436/commits/07712eabcca326013858024b1e2bf963e0c350f0#diff-d8a0e523e68ccf71d1f516f9db119ae5).

nordic-krch commented on src/nrfx/hal/nrf_rng.h:
```diff
@@ -90,98 +90,109 @@ typedef enum
+static inline uint32_t nrf_rng_int_enable_check(NRF_RNG_Type const * p_reg, uint32_t mask)
 {
-    return (bool)(NRF_RNG->INTENSET & rng_int_mask);
+    return p_reg->INTENSET & mask;
```
why p_reg is used here when in other function NRF_RNG is used?

anangl replied:
Because for this one I needed to also update the implementation with the real nrfx HAL one, and I apparently forgot to use `NRF_RNG` as in other functions. Actually, this makes no difference, since there is only one instance of `NRF_RNG`, but for consistency I'll align this function with others.
Thanks for catching this.

anangl replied:
Done.

aescolar commented:
Thank you very much @anangl  , this being a backwards incompatible change is going to be a bit 
troublesome :( . It will give a little bit more of work to not make it too unconvenient for users, but we should be able to manage something nice enough

anangl committed:
nrfx HAL: Align with nrfx version 2.0.0

Update nrfx HAL functions to reflect API changes introduced in nrfx
2.0.0. All these functions are now called with the first parameter
pointing to the structure of registers of the relevant peripheral.
Also a few functions got renamed:
- nrf_rtc_event_pending to nrf_rtc_event_check
- nrf_rtc_int_is_enabled to nrf_rtc_int_enable_check
- nrf_rng_event_get to nrf_rng_event_check
- nrf_rng_int_get to nrf_rng_int_enable_check
- nrf_timer_cc_write to nrf_timer_cc_set
Function nrf_rtc_int_get has been removed.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>change
